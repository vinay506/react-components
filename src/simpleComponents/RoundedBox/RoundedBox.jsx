/**
 *
 * Descirption:
 *   It is provided Rounded Shaped Box with the given height and width from the props
 *  */ 

import React from 'react'
import styled from 'styled-components';

function RoundedBox(props) {
    return (
        <Circle {...props}>
            
        </Circle>
    )
}

const Circle = styled.div`
    background-color: #bbb;
    border-radius: 50%;
    display: inline-block;
`;

export default RoundedBox;
