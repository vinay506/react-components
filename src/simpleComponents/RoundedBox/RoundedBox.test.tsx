import React from 'react';
import { shallow } from 'enzyme';
import RoundedBox from './RoundedBox';
import renderer from 'react-test-renderer'
import 'jest-styled-components'


describe('<Rounded Box>', function () {
    let wrapper:any;
    let props:any;
    let circle:any;
    beforeEach(() => {
        props = {
            style: {
                height: '40px',
                width: '40px',
                marginBottom: '20px'
            }
        };
        wrapper = shallow(<RoundedBox {...props}></RoundedBox>);
        circle = wrapper.dive().find('div');
    });

    // Snapshot test
    it('should render', () => {
        const tree = renderer.create(<RoundedBox {...props}></RoundedBox>).toJSON()
        expect(tree).toMatchSnapshot(`
        .c0 {
              background-color: #bbb;
              border-radius: 50%;
              display: inline-block;
        }
        <div className='sc-AxjAm bfFsoV'
        style={
            className="c0"             
            Object {
                "height": "40px",
                "marginBottom": "20px",
                "width": "40px",
          }
        }></div>`)
    })

    it('renders the Circle Div', () => {
        expect(circle.length).toBe(1);
    });

    it('checks the height', () => {
        expect(circle.props().style.height).toEqual(props.style.height);
    });

    it('checks the width', () => {
        expect(circle.props().style.width).toEqual(props.style.width);
    });

    it('checks the background ', () => {
        const tree = renderer.create(<RoundedBox {...props}></RoundedBox>).toJSON();
        expect(tree).toHaveStyleRule('background-color', '#bbb')
    });

    it('checks the broder radius ', () => {
        const tree = renderer.create(<RoundedBox {...props}></RoundedBox>).toJSON();
        expect(tree).toHaveStyleRule('border-radius', '50%')
    });

})