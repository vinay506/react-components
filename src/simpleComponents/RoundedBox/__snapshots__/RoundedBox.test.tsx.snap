// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rounded Box> should render: 
        .c0 {
              background-color: #bbb;
              border-radius: 50%;
              display: inline-block;
        }
        <div className='sc-AxjAm bfFsoV'
        style={
            className="c0"             
            Object {
                "height": "40px",
                "marginBottom": "20px",
                "width": "40px",
          }
        }></div> 1`] = `
.c0 {
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
}

<div
  className="c0"
  style={
    Object {
      "height": "40px",
      "marginBottom": "20px",
      "width": "40px",
    }
  }
/>
`;
